### DISCUSSION TITLE
Mass Transit & Hotels

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#17470] [Feature requests]
Disclaimer: Old ability of peeps entering the map by foot is of course still available, doesn't affect old scenarios.

Manage the front of your park before the peeps have gone through the entrance.

- Make parks buy land for parking lots, instead of letting people enter the map on foot they could enter by car and would have to park their car in the lot. Lot full, no more peeps can enter the park. 

- Similar to that we can have busses come in on the path, stop at a bus stop and let crowds of like 30-50 peeps come in at the same time.

- Give the ability for transport rides to start at the map edge and leave at the map edge, exiting a station platform and going from that station platform to park entrance.

- Hotels! Manage for people to stay, sell season passes or multiple day passes, get some extra funds from peeps by letting them sleep in the on-site hotel, the peeps refresh and go for another go. Perfect for big parks, so your peeps can explore the entire area.

Some of these features could be scenario editor only (like the transport rides, bus stop and selecting that it's a car entry or foot entry etc)  
Others you could purchase land for in the main game to upgrade your parking lot, or plop down a and on-site hotel. (Designated for this rather than for rides. BUT you can still build a rollercoaster over this land if you want)

I think this could be a really fun expansion to the game.

Let me know your ideas on this.

**Comment by ocalhoun6** [comment]
It would be a _massively_ complex undertaking to code all of this, likely needing new game assets as well. Even more of an undertaking to make sure it doesn't introduce any bugs or have unintended side effects. 

Like ... what happens if the player tries to exploit the system by deleting the path outside of the park once he gets enough peeps in the park, in order to lock in his guest count? Will peep's happiness change and affect the park rating even when they're not inside the park boundary? Will the peeps know which car to return to when leaving, or will they just pick a random one? If they're trying for a specific one, what happens if your parking lot design is bad and other cars have blocked them in?

It's kind of a neat idea... But I suspect the devs need something stronger than 'neat idea' in order to justify implementing so much work. (Especially when you remember that every minute they spend on this is a minute they _don't_ spend on other bugfixes and new features.) It's also kind of infecting the game with scope creep.

Unless there's an unexpectedly huge demand for it in the fanbase or somebody wants to do all the coding themselves ... I suspect this is just a pipe dream. Personally, I don't think it's really necessary. Perfectly adequeate to have a short path to the park entrance and then just assume that all the parking lots and stuff are just outside the edge of the map.

Thinking about that 'off the edge of the map' part, though ... maybe _part_ of this could be implemented -- maybe as a plugin? -- not having parking lots and hotels actually on the map, _but_ having parking lots and hotels be part of the finances window. So players wouldn't actually build them, but they might still have to spend money on improving parking lots/hotels in order to remove the cap on how many peeps could get into the park. Maybe also with the option to charge a parking fee for each guest. Something with no graphical element except for the gui of one or two extra windows/tabs. That could at least get you the _financial_ side of this, without needing anything too complex. The plugin would just have to prevent peeps from spawning unless you've bought enough parking lots for them to park in. Click 'build more parking', and the plugin subtracts money from your balance and increases its artificial guest cap. ... And then _maybe_ that could someday grow into having a visual representation of how many parking spaces you have, and then grow into being able to design/position the parking yourself, etc, etc. That could give you a way to start something smaller and more achievable, and maybe eventually develop it toward what you wanted to see.

**Reply by karst** [reply]
I personally wouldn't see the main devs put their time in this, mainly while OpenRCT2 isn't released. I had the idea for an expansion like this because of talks in discord. I personally think it's worth saving, and perhaps a community member would like to implement this. The idea is as an expansion, i think it'd make for some interesting new scenarios gameplay wise. But similar to most features the core team shouldn't have to focus on this.

**Reply by jjpb44** [reply]
The idea of hotels... Makes me wonder, what about introducing day/night cycles? But this also makes me wonder if the time in the game is too fast, maybe the days could be a current month or so.  On the other hand, presenting all of this can be conflicting, but some of it can be introduced in an add-on or eventually an expansion.

**Reply by James103** [reply]
> But this also makes me wonder if the time in the game is too fast, maybe the days could be a current month or so

This could be done via a day length patch or plug-in, possibly with adjusted capital expense costs to compensate (for long-term games / servers). But that's left to another discussion.

**Reply by ocalhoun6** [reply]
> if the time in the game is too fast, maybe the days could be a current month or so

The game already has fairly reasonable day/night cycles if you turn the lighting effects on.
